src/redbug.erl

```diff
@@ -361,95 +533,39 @@ chk_rtp(Str) when ?is_string(Str)      -> redbug_msc:transform(Str);
 chk_rtp({M})                           -> chk_rtp({M,'_',[]});
 chk_rtp({M,F}) when is_atom(F)         -> chk_rtp({M,F,[]});
 chk_rtp({M,L}) when is_list(L)         -> chk_rtp({M,'_',L});
-chk_rtp({'_',_,_})                     -> exit(dont_wildcard_module);
+chk_rtp({'_',_,_})                     -> throw(dont_wildcard_module);
 chk_rtp({M,F,MS}) when ?is_aal(M,F,MS) -> {{M,F,'_'},ms(MS),[local]};
-chk_rtp(X)                             -> exit({bad_rtp,X}).
+chk_rtp(X)                             -> throw({bad_rtp,X}).
```

Comment:
Around here, there is diff output between this branch and upstream 0.92.0 [1].
It seems like the tuple style RTP had been removed from upstream by the commit [2].
Is it reserved for backward compatibility?


[1]

```
% git diff 0.92.0 ku-merge-upstream-0.92 -- src/redbug.erl
diff --git a/src/redbug.erl b/src/redbug.erl
index e7dbf01..46560f8 100644
--- a/src/redbug.erl
+++ b/src/redbug.erl
@@ -527,7 +527,26 @@ chk_trc('receive',{Flags,RTPs})                -> {['receive'|Flags],RTPs};
 chk_trc(RTP,{Flags,RTPs}) when ?is_string(RTP) -> {Flags,[chk_rtp(RTP)|RTPs]};
 chk_trc(X,_)                                   -> throw({bad_trc,X}).

-chk_rtp(Str) -> redbug_msc:transform(Str).
+-define(is_aal(M,F,MS), is_atom(M),is_atom(F),is_list(MS)).
+
+chk_rtp(Str) when ?is_string(Str)      -> redbug_msc:transform(Str);
+chk_rtp({M})                           -> chk_rtp({M,'_',[]});
+chk_rtp({M,F}) when is_atom(F)         -> chk_rtp({M,F,[]});
+chk_rtp({M,L}) when is_list(L)         -> chk_rtp({M,'_',L});
+chk_rtp({'_',_,_})                     -> throw(dont_wildcard_module);
+chk_rtp({M,F,MS}) when ?is_aal(M,F,MS) -> {{M,F,'_'},ms(MS),[local]};
+chk_rtp(X)                             -> throw({bad_rtp,X}).
+
+ms(MS) -> lists:foldl(fun msf/2, [{'_',[],[]}], MS).
+
+msf(stack,[{Head,Cond,Body}]) -> [{Head,Cond,[{message,{process_dump}}|Body]}];
+msf(return,[{Head,Cond,Body}])-> [{Head,Cond,[{exception_trace}|Body]}];
+msf(Ari, [{_,Cond,Body}]) when is_integer(Ari)-> [{mk_head(Ari),Cond,Body}];
+msf({Head,Cond},[{_,_,Body}]) when is_tuple(Head)->[{Head,slist(Cond),Body}];
+msf(Head, [{_,Cond,Body}]) when is_tuple(Head)-> [{Head,Cond,Body}];
+msf(X,_) -> throw({bad_match_spec,X}).
+
+mk_head(N) -> erlang:make_tuple(N,'_').

 slist(S) when ?is_string(S) -> [S];
 slist(L) when is_list(L) -> lists:usort(L);
```

[2] remove old tuple-API from redbug · massemanet/eper@fa935b2
  https://github.com/massemanet/eper/commit/fa935b2c6c289e7ac0c9728c090b66ef506724f6?w=1